crates/examples/infra/mod.rs

```diff
@@ -1027,10 +1042,42 @@ pub async fn main_entry_point<
         .await
         .expect("failed to get config");
 
+    let builder_task = match run_config.builder {
+        BuilderType::External => None,
+        BuilderType::Random => {
+            let (builder_task, builder_url) =
+                <RandomBuilderImplementation as TestBuilderImplementation<TYPES>>::start(
+                    run_config.config.num_nodes_with_stake.into(),
+                    run_config.random_builder.clone().unwrap_or_default(),
+                )
+                .await;
+
+            run_config.config.builder_url = builder_url;
+
+            builder_task
+        }
+        BuilderType::Simple => {
+            let (builder_task, builder_url) =
+                <SimpleBuilderImplementation as TestBuilderImplementation<TYPES>>::start(
+                    run_config.config.num_nodes_with_stake.into(),
+                    (),
+                )
+                .await;
+
+            run_config.config.builder_url = builder_url;
+
+            builder_task
+        }
+    };
+
     info!("Initializing networking");
     let run = RUNDA::initialize_networking(run_config.clone(), args.advertise_address).await;
     let hotshot = run.initialize_state_and_hotshot().await;
 
+    if let Some(task) = builder_task {
+        task.start(Box::new(hotshot.get_event_stream()));
```

Comment:
Does this run a builder for each node? If so, I think instead we should consider a separate (simple) binary that just runs the simple builder. Then in `all.rs` we could add a call to start the simple builder, and when we have large, multi-node runs, we can run the binary

Reply:
Yeah, it runs a builder for each node.
Running a single builder per network would probably make more sense, honestly. WDYT about adding a cmdline/config option for amount of builders to run and defaulting it to 1?
We need to start the simple builder together with a hotshot node, because it (for now?) relies on hs event stream to gather transactions.

Reply:
Ah right, yes this makes sense to do but only if it's not too much of a change. It shouldn't matter _too much_ because we are generally doing small runs with `all.rs`.

For benchmarking runs, we will need a builder binary, but assuming that's out of scope for this one

Reply:
> Ah right, yes this makes sense to do but only if it's not too much of a change. It shouldn't matter too much because we are generally doing small runs with all.rs.

Yeah, it shouldn't be too much work, I'll add it today.

> For benchmarking runs, we will need a builder binary, but assuming that's out of scope for this one

Any explicit support for external builder binary is out of scope for this, I'm going to discuss how best to do it with builder team after this one.

Reply:
Self-quoting over from Zulip
> Meanwhile, I realised that properly supporting running arbitrary number of integrated builders with examples is a bit tricky, since there needs to be some back-and-forth with orchestrator to properly assign builder addresses to nodes. Do you mind if we merge #2935 as-is for now, and I'll open a separate issue to allow running arbitrary number of integrated builders?